src/assisted_test_infra/test_infra/helper_classes/cluster.py

```diff
@@ -127,11 +127,14 @@ def _create(self) -> str:
         return cluster.id
 
     def delete(self):
+        self.deregister_infraenv()
         self.api_client.delete_cluster(self.id)
+        self._config.cluster_id = None
```

Comment:
Locally teardown is working (and also on CI). 
Maybe you are simulating another scenario?

Reply:
Probably. Scenario is create cluster1, boot nodes, delete cluster1 and nodes, create cluster2 with nodes, boot nodes, than teardown: https://gitlab.cee.redhat.com/ocp-edge-qe/kni-assisted-installer-auto/-/blob/master/api_tests/test_dhcp.py#L66
So i think we need to update cluster config itself i.e `cluster_config.cluster_id = None` and not the `self._config.cluster_id = None `

Reply:
Writing this here so it will be documented.
You cannot pass the same `ClusterConfig` to the second cluster, it will cause unwanted behavior due to that the configurations object is mutable, so in that case, teardown is deleting the same cluster twice (both clusters will have cluster2 id).
Also I added a check before deleting a cluster so it will not try to delete the cluster if it's ID is None